eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/AsyncPaymentTriggerer.scala

```diff
@@ -0,0 +1,127 @@
+    def expired(currentBlockHeight: BlockHeight): Boolean = timeout <= currentBlockHeight
+  }
+
+  case class PeerPayments(notifier: ActorRef[PeerReadyNotifier.Command], pendingPayments: Set[Payment]) {
+    def update(currentBlockHeight: BlockHeight): Option[PeerPayments] = {
+      val expiredPayments = pendingPayments.filter(_.expired(currentBlockHeight))
+      expiredPayments.foreach(e => e.replyTo ! AsyncPaymentTimeout)
+      val pendingPayments1 = pendingPayments.removedAll(expiredPayments)
+      if (pendingPayments1.isEmpty) {
+        context.stop(notifier)
+        None
+      } else {
+        Some(PeerPayments(notifier, pendingPayments1))
+      }
+    }
+
+    def trigger(): Unit = pendingPayments.foreach(e => e.replyTo ! AsyncPaymentTriggered)
+    def timeout(): Unit = pendingPayments.foreach(e => e.replyTo ! AsyncPaymentTimeout)
+  }
+
+  def start(): Behavior[Command] = {
+    context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[CurrentBlockHeight](WrappedCurrentBlockHeight))
+    watching(Map.empty)
+  }
+
+  private def watching(peers: Map[PublicKey, PeerPayments]): Behavior[Command] = {
+    Behaviors.receiveMessagePartial {
+      case Watch(replyTo, remoteNodeId, paymentHash, timeout) =>
+        peers.get(remoteNodeId) match {
+          case None =>
+            val notifier = context.spawn(
+              Behaviors.supervise(PeerReadyNotifier(remoteNodeId, switchboard, None)).onFailure(SupervisorStrategy.stop),
+              s"peer-ready-notifier-$remoteNodeId",
+            )
+            context.watchWith(notifier, NotifierStopped(remoteNodeId))
+            notifier ! NotifyWhenPeerReady(context.messageAdapter[PeerReadyNotifier.Result](WrappedPeerReadyResult))
+            val peer = PeerPayments(notifier, Set(Payment(replyTo, timeout, paymentHash)))
+            watching(peers + (remoteNodeId -> peer))
+          case Some(peer) =>
+            val peer1 = PeerPayments(peer.notifier, peer.pendingPayments + Payment(replyTo, timeout, paymentHash))
+            watching(peers + (remoteNodeId -> peer1))
+        }
+      case WrappedCurrentBlockHeight(CurrentBlockHeight(currentBlockHeight)) =>
+        val peers1 = peers.flatMap {
+          case (remoteNodeId, peer) => peer.update(currentBlockHeight).map(peer1 => remoteNodeId -> peer1)
+        }
+        watching(peers1)
+      case WrappedPeerReadyResult(PeerReadyNotifier.PeerReady(remoteNodeId, _)) =>
+        // notify watcher that destination peer is ready to receive async payments; PeerReadyNotifier will stop itself
+        peers.get(remoteNodeId).foreach(_.trigger())
+        watching(peers - remoteNodeId)
+      case NotifierStopped(remoteNodeId) =>
+        peers.get(remoteNodeId) match {
+          case None => Behaviors.same
+          case Some(peer) => peer.timeout()
+            context.log.error(s"PeerReadyNotifier stopped unexpectedly while watching node $remoteNodeId.")
+            watching(peers - remoteNodeId)
```

Comment:
nit: when using a multi-line block of code, put all statements in the same block:

```suggestion
          case Some(peer) =>
            context.log.error(s"PeerReadyNotifier stopped unexpectedly while watching node $remoteNodeId.")
            peer.timeout()
            watching(peers - remoteNodeId)
```